### ISSUE TITLE
Unused type parameter server vs. command-line behaviour

### ISSUE THREAD

**Issue by tpgillam** [microsoft/pyright#10155]
Consider the following code, in which the `T` type parameter is unused:

```python
# moo.py
class Moo[T]:
    pass
```

In pyright 1.1.396 (tested in both "basic" and "strict" modes), this code does not trigger any warnings or errors when running `pyright moo.py`.

But, if running pyright as a language server over this file, then we see a diagnostic remark from pyright:
```python
class Moo[T]:     ■ "T" is not accessed
    pass
```

It would be nice if this remark/warning were detectable when run from the command line too.

Should this remark flag one of the existing checks? Or would it require adding a new check?
